compute_endpoint/tests/unit/test_endpoint_unit.py

```diff
@@ -263,39 +264,73 @@ def test_register_endpoint_invalid_response(
     assert other_endpoint_id in mock_log.error.call_args[0][0]
 
 
-@pytest.mark.parametrize("ret_value", [[409, "Conflict"], [423, "Locked"]])
+# @pytest.mark.parametrize("ret_value", [[409, "Conflict"], [423, "Locked"]])
+@pytest.mark.parametrize(
+    "status_code",
+    [
+        HTTPStatus.BAD_REQUEST,
+        HTTPStatus.NOT_FOUND,
+        HTTPStatus.CONFLICT,
+        HTTPStatus.UNPROCESSABLE_ENTITY,
+        HTTPStatus.LOCKED,
+        HTTPStatus.IM_A_TEAPOT,
+    ],
+)
 @responses.activate
-def test_register_endpoint_locked_conflict_print(
+def test_register_endpoint_blocked(
     mocker,
     fs,
     register_endpoint_failure_response,
     get_standard_compute_client,
     mock_ep_data,
-    ret_value,
+    randomstring,
+    status_code,
 ):
     """
     Check to ensure endpoint registration escalates up with API error
     """
-    ret_code, ret_text = ret_value
+    mock_log = mocker.patch(f"{_mock_base}log")
     mock_gcc = get_standard_compute_client()
     mocker.patch(f"{_mock_base}Endpoint.get_funcx_client").return_value = mock_gcc
     f = io.StringIO()
 
     ep, ep_dir, log_to_console, no_color, ep_conf = mock_ep_data
     ep_id = str(uuid.uuid4())
+    some_err = randomstring()
     register_endpoint_failure_response(
         endpoint_id=ep_id,
-        status_code=ret_code,
-        msg=ret_text,
+        status_code=status_code,
+        msg=some_err,
     )
+
     with redirect_stdout(f):
-        with pytest.raises(SystemExit) as pytest_exc:
+        with pytest.raises((GlobusAPIError, SystemExit)) as pytexc:
             ep.start_endpoint(
                 ep_dir, ep_id, ep_conf, log_to_console, no_color, reg_info={}
             )
-        err_msg = f.getvalue()
-        assert "Endpoint registration blocked" in err_msg and ret_text in err_msg
-        assert pytest_exc.value.code == os.EX_UNAVAILABLE
+        stdout_msg = f.getvalue()
+
+    assert mock_log.warning.called
+    a, *_ = mock_log.warning.call_args
+    assert some_err in str(a), "Expected upstream response still shared"
+
+    if status_code in (
+        HTTPStatus.CONFLICT,
+        HTTPStatus.LOCKED,
+        HTTPStatus.NOT_FOUND,
+    ):
+        assert some_err in stdout_msg, "Expecting error message in stdout"
+        assert pytexc.value.code == os.EX_UNAVAILABLE, "Expecting meaningful exit code"
+    elif status_code in (
+        HTTPStatus.BAD_REQUEST,
+        HTTPStatus.UNPROCESSABLE_ENTITY,
+    ):
+        assert some_err in stdout_msg, "Expecting error message in stdout"
+        assert pytexc.value.code == os.EX_DATAERR, "Expecting meaningful exit code"
```

Comment:
From a test organization perspective, suggest instead to combine these at parametrization-time.  Consider:
```python
@pytest.mark.parametrize(
    "exit_code,status_code",
    (
        (os.EX_UNAVAILABLE, HTTPStatus.CONFLICT),
        (os.EX_UNAVAILABLE, HTTPStatus.LOCKED),
        (os.EX_UNAVAILABLE, HTTPStatus.NOT_FOUND),
        (os.EX_DATAERR, HTTPStatus.BAD_REQUEST),
        (os.EX_DATAERR, HTTPStatus.UNPROCESSABLE_ENTITY),
        ("Error", HTTPStatus.IM_A_TEAPOT),
    ),
)
```
... which, in combination of the previous `print()` statement suggestion, would allow the removal of the multi-branch if in this test:
```diff
-    if status_code in (
-        HTTPStatus.CONFLICT,
-        HTTPStatus.LOCKED,
-        HTTPStatus.NOT_FOUND,
-    ):
-        assert some_err in stdout_msg, "Expecting error message in stdout"
-        assert pytexc.value.code == os.EX_UNAVAILABLE, "Expecting meaningful exit code"
-    elif status_code in (
-        HTTPStatus.BAD_REQUEST,
-        HTTPStatus.UNPROCESSABLE_ENTITY,
-    ):
-        assert some_err in stdout_msg, "Expecting error message in stdout"
-        assert pytexc.value.code == os.EX_DATAERR, "Expecting meaningful exit code"
-    else:
+    assert some_err in stdout_msg, f"Expecting error message in stdout ({stdout_msg})"
+    assert pytexc.value.code == exit_code, "Expecting meaningful exit code"
+
+    if exit_code == "Error":
```

Reply:
Very good suggestion.